### PR TITLE
Add docstrings linting with 'pydocstyle' using 'flake8'

### PR DESCRIPTION
**Pull request by mondeja**
This pull request add configuration and installation for [`flake8-docstrings`](https://gitlab.com/pycqa/flake8-docstrings) and [`flake8-rst-docstrings`](https://github.com/peterjc/flake8-rst-docstrings). It ensures that docstring haven't common errors, that docstrings are written for public functions... unifying in general the code base.

I've ignored some rules either because it generated too much work to correct them (`D100`, `D104`) or are too much intrusive (`D401`, for example).

Other minor changes are:

- Removed empty module `positioning.py`.
- Removed unused function `PIL_to_npimage`.

### CODE REVIEW

File: moviepy/utils.py [hunk 1]
```diff
@@ -11,6 +11,25 @@
...
+    """Closes all clips in a context.
+
+    Follows different strategies retrieving the namespace from which the clips
+    to close will be retrieved depending on the ``objects`` argument, and filtering
+    by type of clips depending on the ``types`` argument.
+
+    Parameters
+    ----------
+
+    objects : str/dict, optional
```
Comment (tburrows13) [inline moviepy/utils.py:23]:
```suggestion
    objects : str or dict, optional
```
According to https://numpydoc.readthedocs.io/en/latest/format.html#sections :)

File: moviepy/Clip.py [hunk 2]
```diff
@@ -91,22 +81,22 @@ def transform(self, func, apply_to=None, keep_duration=True):
...
-        func
+        func : function
           A function with signature (gf,t -> frame) where ``gf`` will
           represent the current clip's ``get_frame`` method,
           i.e. ``gf`` is a function (t->image). Parameter `t` is a time
           in seconds, `frame` is a picture (=Numpy array) which will be
           returned by the transformed clip (see examples below).
 
-        apply_to
+        apply_to : Iterable
```
Comment (tburrows13) [inline moviepy/Clip.py:93]:
```suggestion
        apply_to : {"mask", "audio", ["mask", "audio"]}
```

File: moviepy/Clip.py [hunk 3]
```diff
@@ -151,17 +141,17 @@ def time_transform(self, time_func, apply_to=None, keep_duration=False):
...
-        -----------
+        ----------
 
-        time_func:
-          A function ``t -> new_t``
+        time_func : function
+          A function ``t -> new_t``.
 
-        apply_to:
+        apply_to : Iterable
```
Comment (tburrows13) [inline moviepy/Clip.py:149]:
```suggestion
        apply_to : {"mask", "audio", ["mask", "audio"]}
```

File: moviepy/video/fx/resize.py [hunk 4]
```diff
@@ -116,36 +116,33 @@ def _get_resizer():
...
-    """
-    Returns a video clip that is a resized version of the clip.
+    """Returns a video clip that is a resized version of the clip.
 
     Parameters
-    ------------
+    ----------
 
-    new_size:
+    new_size : tuple, optional
```
Comment (tburrows13) [inline moviepy/video/fx/resize.py:124]:
```suggestion
    new_size : tuple or float or function, optional
```

File: moviepy/video/io/VideoFileClip.py [hunk 5]
```diff
@@ -148,9 +146,11 @@ def mask_make_frame(t):
             )
 
     def __deepcopy__(self, memo):
-        """AudioFileClip can't be deeply copied because the locked Thread
+        """Implements ``copy.deepcopy(clip)`` behaviour as ``copy.copy(clip)``.
+
+        VideoFileClip class instances can't be deeply copied because the locked Thread
         of ``proc`` isn't pickleable. Without this override, calls to
-        ``copy.deepcopy(clip)`` will raise next ``TypeError``:
+        ``copy.deepcopy(clip)`` would raise next ``TypeError``:
```
Comment (tburrows13) [inline moviepy/video/io/VideoFileClip.py:153]:
```suggestion
        ``copy.deepcopy(clip)`` would raise a ``TypeError``:
```
"will" or "would" are both fine here, but I don't think "next" makes any sense?

File: moviepy/video/io/ffmpeg_tools.py [hunk 6]
```diff
@@ -1,4 +1,4 @@
-""" Misc. bindings to ffmpeg and ImageMagick."""
+"""Miscelaneous bindings to ffmpeg."""
```
Comment (tburrows13) [inline moviepy/video/io/ffmpeg_tools.py:1]:
```suggestion
"""Miscellaneous bindings to ffmpeg."""
```

File: moviepy/video/tools/credits.py [hunk 7]
```diff
@@ -10,86 +8,85 @@
...
-        color="white",
-        stroke_color="black",
-        stroke_width=2,
-        font="Impact-Normal",
-        font_size=60,
-        bg_color=None,
-        gap=0,
-    ):
-        """
+    """Experimental credits clip.
```
Comment (tburrows13) [inline moviepy/video/tools/credits.py:11]: Why is it "Experimental"?

**Review by tburrows13 — COMMENTED** [review]
Very nice overall. This should make my job in #1482 easier (after a painful rebase/merge!). The only other thing I'm not sure about is indenting examples. I don't think it is necessary, but I think you've done that here? If I'm wrong, please ignore me!

File: moviepy/video/tools/credits.py (hunk 7 above)
Comment (mondeja) [reply moviepy/video/tools/credits.py:11]: I don't know :sweat_smile: Let me change it

**Comment by mondeja** [comment]
> The only other thing I'm not sure about is indenting examples. I don't think it is necessary, but I think you've done that here? 

Yes, I've found indented the examples of a function, but is fixed now. Let me know if you see something more wrong.

**Review by tburrows13 — APPROVED** [review]

**Pull request merged**